lib/request.js

```diff
@@ -319,7 +319,14 @@ module.exports = {
     var type = this.get('Content-Type');
     if (!type) return '';
 
-    return contentType.parse(type).parameters.charset || '';
+    try {
+      type = contentType.parse(type);
+    } catch (e) {
+      if (e instanceof TypeError) return '';
+      throw e
```

Comment:
maybe we can remove `if (e instanceof TypeError)` here since this line will never hit.

others LGTM